### PR TITLE
Request manager whitelist

### PR DESCRIPTION
**Pull request by fredo**
resolves: #893 

For the tests I created a function called `alloc_whitelist_accounts` which takes contracts as an argument and calls `addAllowedLp()` on it. Since it is a util function in must be passed in the arguments.
The positive side on this is that I had to go through all the tests and bring more context to the addresses and give them better names. I also removed some overlappings where the requester was also the claimer (not that it is harmful in particular but it removes the possibgility of failures due to address misusage).

Regarding `call_contracts.py whitelist` I did not come up with a proper solution because the script only consumes one RPC endpoint but it would require two here. Since it is not critical for the audit I suggest to create a follow up issue for that.

### CODE REVIEW

File: beamer/tests/util.py [hunk 1]
```diff
@@ -28,6 +29,14 @@ def alloc_accounts(n):
     return tuple(_alloc_account() for _ in range(n))
 
 
+def alloc_whitelisted_accounts(n, contracts):
```
Comment (cducrest) [inline beamer/tests/util.py:32]: This assumes each contracts is owned by the default address. Maybe that's an assumption you have everywhere in the tests, in which case it is fine.

File: beamer/tests/contracts/test_fill_manager.py [hunk 2]
```diff
@@ -2,28 +2,32 @@
...
     with brownie.reverts("Ownable: caller is not the owner"):
-        fill_manager.addAllowedLP(deployer, {"from": filler})
+        fill_manager.addAllowedLp(filler, {"from": receiver})
 
-    whitelist_tx = fill_manager.addAllowedLP(deployer, {"from": deployer})
-    assert "LPAdded" in whitelist_tx.events
+    whitelist_tx = fill_manager.addAllowedLp(receiver)
+    assert "LpAdded" in whitelist_tx.events
+    blacklist_tx = fill_manager.removeAllowedLp(receiver)
+    assert "LpRemoved" in blacklist_tx.events
```
Comment (cducrest) [inline beamer/tests/contracts/test_fill_manager.py:20]: It's boring to read tests that do a little bit of everything, but this PR does not change it.

File: beamer/tests/contracts/test_request_manager.py [hunk 3]
```diff
@@ -40,15 +41,22 @@ def test_request_invalid_target_chain(request_manager, token):
...
+    (claimer,) = alloc_whitelisted_accounts(1, {request_manager})
     request_id = make_request(
         request_manager, token=token, requester=requester, target_address=requester, amount=1
     )
 
+    with brownie.reverts("Ownable: caller is not the owner"):
+        request_manager.addAllowedLp(claimer, {"from": requester})
+
+    whitelist_tx = request_manager.addAllowedLp(claimer, {"from": deployer})
+    assert "LpAdded" in whitelist_tx.events
```
Comment (cducrest) [inline beamer/tests/contracts/test_request_manager.py:56]: Same as previous comment, this has little to do with `test_claim`

**Review by cducrest — APPROVED** [review]

File: beamer/tests/contracts/test_fill_manager.py (hunk 2 above)
Comment (fredo) [reply beamer/tests/contracts/test_fill_manager.py:20]: Yes, I'm also not happy about that we have some tests which do a little bit of everything. That's also the case on lots of other places, but I totally agree. 

The lines you mentioned add and remove the `receiver` though not the filler, in order to test that events are created. Still misplaced.

Comment (fredo) [reply beamer/tests/contracts/test_fill_manager.py:20]: We should definitely refactor this

**Pull request merged**